src/game.cpp

```diff
@@ -7141,6 +7141,16 @@ void game::smash()
             return; // don't smash terrain if we've smashed a corpse
         }
     }
+    bool ter_or_furn = m.has_flag_ter_or_furn( "ELECTRIFIED", smashx, smashy );
+    if (ter_or_furn) {
+            body_part bp = random_body_part();
+            if(u.deal_damage( nullptr, bp, damage_instance( DT_ELECTRIC, rng( 5, 20 ) ) ).total_damage() > 0) {
+                add_msg(m_bad, _("You are shocked by the %1$s!"),
+                    ter_or_furn ? m.tername(smashx, smashy).c_str() : m.furnname(smashx, smashy).c_str());
```

Comment:
I think this also helps identify a bug outside this PR in game.cpp. Line 11996 and the block immediately after this relating to sharp terrain/furniture.

```C++
        if (m.has_flag("ROUGH", x, y) && (!u.in_vehicle)) {
            bool ter_or_furn = m.has_flag_ter_or_furn( "ROUGH", x, y );
            if (one_in(5) && u.get_armor_bash(bp_foot_l) < rng(2, 5)) {
                add_msg(m_bad, _("You hurt your left foot on the %s!"),
                        ter_or_furn ? m.tername(x, y).c_str() : m.furnname(x, y).c_str() );
                u.deal_damage( nullptr, bp_foot_l, damage_instance( DT_CUT, 1 ) );
            }
            if (one_in(5) && u.get_armor_bash(bp_foot_r) < rng(2, 5)) {
                add_msg(m_bad, _("You hurt your right foot on the %s!"),
                        ter_or_furn ? m.tername(x, y).c_str() : m.furnname(x, y).c_str() );
                u.deal_damage( nullptr, bp_foot_l, damage_instance( DT_CUT, 1 ) );
            }
        }
```

ter_or_furn should be m.ter_has_flag here as well should it not?